yices2/include/yices2_solver.h

```diff
@@ -0,0 +1,94 @@
+#ifndef SMT_YICES2_SOLVER_H
+#define SMT_YICES2_SOLVER_H
+
+#include <gmp.h>
+#include <memory>
+#include <string>
+#include <unordered_set>
+#include <vector>
+
+#include <yices.h>
+
+#include "yices2_extensions.h"
+#include "yices2_sort.h"
+#include "yices2_term.h"
+
+#include "exceptions.h"
+#include "result.h"
+#include "smt.h"
+#include "sort.h"
+
+namespace smt {
+/**
+   Yices2 Solver
+ */
+class Yices2Solver : public AbsSmtSolver
+{
+ public:
+  Yices2Solver()
+  {
+    // Had to move yices_init to the Factory
+    // yices_init();
+    ctx = yices_new_context(NULL);
+    config = yices_new_config();
+  };
+  Yices2Solver(const Yices2Solver &) = delete;
+  Yices2Solver & operator=(const Yices2Solver &) = delete;
+  ~Yices2Solver()
+  {
+    yices_free_config(config);
+    yices_free_context(ctx);
+
+    // TODO: Should probably find a good place to
+    // call yices_exit.
+    // yices_exit();
+  };
+  void set_opt(const std::string option, const std::string value) override;
+  void set_logic(const std::string logic) const override;
```

Comment:
Hey, sorry I had changed this in a recent PR. Could you only have one `set_logic` and not make it `const`? Then I think we should be good to go -- Travis almost completely passed this time haha.

Reply:
Thanks for the change @amaleewilso! I think it needs to be changed here also though, right? https://github.com/makaimann/smt-switch/blob/6c1b2f7eddce2be40f8d14fc4527ec4cdeb44acb/yices2/src/yices2_solver.cpp#L118

Reply:
Thanks, should be fixed.